grid2op/Environment/MultiMixEnv.py

```diff
@@ -0,0 +1,218 @@
+# Copyright (c) 2019-2020, RTE (https://www.rte-france.com)
+# See AUTHORS.txt
+# This Source Code Form is subject to the terms of the Mozilla Public License, version 2.0.
+# If a copy of the Mozilla Public License, version 2.0 was not distributed with this file,
+# you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+# This file is part of Grid2Op, Grid2Op a testbed platform to model sequential decision making in power systems.
+
+import os
+import numpy as np
+
+from grid2op.dtypes import dt_int, dt_float
+from grid2op.Space import GridObjects, RandomObject
+from grid2op.Exceptions import EnvError 
+
+class MultiMixEnvironment(GridObjects, RandomObject):
+    """
+    This class represent a single powergrid configuration,
+    backed by multiple enviromnents parameters and chronics
+
+    It implements most of the BaseEnv public interface:
+    so it can be used as a more classic environment.
+
+    """
+    def __init__(self,
+                 envs_dir):
+        GridObjects.__init__(self)
+        RandomObject.__init__(self)
+
+        self.current_env = None
+        self._envs = []
+
+        # Inline import to prevent cyclical import
+        from grid2op.MakeEnv.Make import make
+
+        try:
+            for env_dir in sorted(os.listdir(envs_dir)):
+                env_path = os.path.join(envs_dir, env_dir)            
+                if not os.path.isdir(env_path):
+                    continue
+                env = make(env_path)
+                self._envs.append(env)
+        except Exception as e:
+            err_msg = "MultiMix environment creation failed: {}".format(e)
+            raise EnvError(err_msg)
+
+        if len(self._envs) == 0:
+            err_msg = "MultiMix envs_dir did not contain any valid env"
+            raise EnvError(err_msg)
+            
+        self.current_env = self._envs[0]
+        # Make sure GridObject class attributes are set from first env
+        # Shouldbe fine since the grid is the same for all envs
+        self.__class__ = self.init_grid(self.current_env)
+
+    def __getattr__(self, name):
+        return getattr(self.current_env, name)
+
+    def reset(self):
+        rnd_env_idx = np.random.randint(len(self._envs))
```

Comment:
Maybe just loop through the _envs deterministically and add a possibility to loop randomly. More control is always better.